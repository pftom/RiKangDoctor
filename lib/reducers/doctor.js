Object.defineProperty(exports,"__esModule",{value:true});var _immutable=require('immutable');var _immutable2=_interopRequireDefault(_immutable);


var _constants=require('../constants/');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}
















var initialDoctorValue=_immutable2.default.Map({
doctorInfo:null,
answers:null,
comments:null,
isLoadingData:false,
loadingError:false,
loadingSuccess:false});


var doctor=function doctor(){var state=arguments.length>0&&arguments[0]!==undefined?arguments[0]:initialDoctorValue;var action=arguments[1];
switch(action.type){
case _constants.GET_SINGLE_DOCTOR_INFO:
case _constants.GET_SINGLE_DOCTOR_ANSWERS:
case _constants.GET_SINGLE_DOCTOR_COMMENTS:

return state.merge({
isLoadingData:true,
loadingError:false,
loadingSuccess:false});


case _constants.GET_SINGLE_DOCTOR_INFO_SUCCESS:var

doctorInfo=action.doctorInfo;
return state.merge({
isLoadingData:false,
loadingSuccess:true,
doctorInfo:doctorInfo});


case _constants.GET_SINGLE_DOCTOR_ANSWERS_SUCCESS:var

answers=action.answers;
return state.merge({
isLoadingData:false,
loadingSuccess:true,
answers:answers});



case _constants.GET_SINGLE_DOCTOR_COMMENTS_SUCCESS:var

comments=action.comments;
return state.merge({
isLoadingData:false,
loadingSuccess:true,
comments:comments});




case _constants.GET_SINGLE_DOCTOR_INFO_ERROR:
case _constants.GET_SINGLE_DOCTOR_ANSWERS_ERROR:
case _constants.GET_SINGLE_DOCTOR_COMMENTS_ERROR:

return state.merge({
isLoadingData:false,
loadingError:true});


default:
return state;}

};exports.default=

doctor;