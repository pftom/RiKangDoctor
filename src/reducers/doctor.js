import Immutable from 'immutable';

//import action constants
import { 
  GET_SINGLE_DOCTOR_INFO,
  GET_SINGLE_DOCTOR_INFO_SUCCESS,
  GET_SINGLE_DOCTOR_INFO_ERROR, 

  GET_SINGLE_DOCTOR_ANSWERS,
  GET_SINGLE_DOCTOR_ANSWERS_SUCCESS,
  GET_SINGLE_DOCTOR_ANSWERS_ERROR,

  GET_SINGLE_DOCTOR_COMMENTS,
  GET_SINGLE_DOCTOR_COMMENTS_SUCCESS,
  GET_SINGLE_DOCTOR_COMMENTS_ERROR,
} from '../constants/';

import {
  combine,
  refreshIt,
} from './utils/';


//home reducers
const initialDoctorValue = Immutable.Map({
  doctorInfo: null,
  answers: null,
  comments: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: false,
});

const doctor = (state = initialDoctorValue, action) => {
  switch (action.type) {
    case GET_SINGLE_DOCTOR_INFO:
    case GET_SINGLE_DOCTOR_ANSWERS:
    case GET_SINGLE_DOCTOR_COMMENTS:

      return state.merge({
        isLoadingData: true,
        loadingError: false,
        loadingSuccess: false,
      });
    
    case GET_SINGLE_DOCTOR_INFO_SUCCESS:

      const { doctorInfo } = action;
      return state.merge({
        isLoadingData: false,
        loadingSuccess: true,
        doctorInfo,
      });
    
    case GET_SINGLE_DOCTOR_ANSWERS_SUCCESS:

      const { answers } = action;

      let oldAnswers = state.get('answers');

      if (answers) {
        answers = Immutable.Map(answers);
      }
      return state.merge({
        isLoadingData: false,
        loadingSuccess: true,
        answers: action.refresh ? refreshIt(oldAnswers, answers) : combine(oldAnswers, answers),
      });


    case GET_SINGLE_DOCTOR_COMMENTS_SUCCESS:

      const { comments } = action;

      let oldComments = state.get('comments');

      if (comments) {
        comments = Immutable.Map(comments);
      }
      return state.merge({
        isLoadingData: false,
        loadingSuccess: true,
        comments: action.refresh ? refreshIt(oldComments, comments) : combine(oldComments, comments),
      });
      
      

    case GET_SINGLE_DOCTOR_INFO_ERROR:
    case GET_SINGLE_DOCTOR_ANSWERS_ERROR:
    case GET_SINGLE_DOCTOR_COMMENTS_ERROR:

      return state.merge({
        isLoadingData: false,
        loadingError: true,
      });

    default:
      return state;
  }
};

export default doctor;