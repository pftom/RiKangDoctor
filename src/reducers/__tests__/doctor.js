import Immutable from 'immutable';

//import test reducer
import doctor from '../doctor';


//import action constants
import { 
  GET_SINGLE_DOCTOR_INFO,
  GET_SINGLE_DOCTOR_INFO_SUCCESS,
  GET_SINGLE_DOCTOR_INFO_ERROR, 

  GET_SINGLE_DOCTOR_ANSWERS,
  GET_SINGLE_DOCTOR_ANSWERS_SUCCESS,
  GET_SINGLE_DOCTOR_ANSWERS_ERROR,

  GET_SINGLE_DOCTOR_COMMENTS,
  GET_SINGLE_DOCTOR_COMMENTS_SUCCESS,
  GET_SINGLE_DOCTOR_COMMENTS_ERROR,
} from '../../constants/'

const initialDoctorValue = Immutable.Map({
  doctorInfo: null,
  answers: null,
  comments: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: false,
});


//test singleQA action
const getDoctorInfoAction = {
  type: GET_SINGLE_DOCTOR_INFO,
};

const getDoctorInfoSuccessAction = {
  type: GET_SINGLE_DOCTOR_INFO_SUCCESS,
  doctorInfo: {}
}

const getDoctorInfoErrorAction = {
  type: GET_SINGLE_DOCTOR_INFO_ERROR,
}

const getDoctorAnswersAction = {
  type: GET_SINGLE_DOCTOR_ANSWERS,
};

const getDoctorAnswersSuccessAction = {
  type: GET_SINGLE_DOCTOR_ANSWERS_SUCCESS,
  answers: {},
};

const getDoctorAnswersErrorAction = {
  type: GET_SINGLE_DOCTOR_ANSWERS_ERROR,
};

const getDoctorCommentsAction = {
  type: GET_SINGLE_DOCTOR_COMMENTS,
};

const getDoctorCommentsSuccessAction = {
  type: GET_SINGLE_DOCTOR_COMMENTS_SUCCESS,
  comments: {},
};

const getDoctorCommentsErrorAction = {
  type: GET_SINGLE_DOCTOR_COMMENTS_ERROR,
};

test('the answer reducer work as well', () => {
  //test doctor info
  expect(doctor(initialDoctorValue, getDoctorInfoAction)).toMatchSnapshot();
  expect(doctor(initialDoctorValue.merge({ isLoadingData: true }), getDoctorInfoSuccessAction)).toMatchSnapshot();
  expect(doctor(initialDoctorValue.merge({ isLoadingData: true }), getDoctorInfoErrorAction)).toMatchSnapshot();

  //test answers
  expect(doctor(initialDoctorValue, getDoctorAnswersAction)).toMatchSnapshot();
  expect(doctor(initialDoctorValue.merge({ isLoadingData: true }), getDoctorAnswersSuccessAction)).toMatchSnapshot();
  expect(doctor(initialDoctorValue.merge({ isLoadingData: true }), getDoctorAnswersErrorAction)).toMatchSnapshot();

  //test comments
  expect(doctor(initialDoctorValue, getDoctorCommentsAction)).toMatchSnapshot();
  expect(doctor(initialDoctorValue.merge({ isLoadingData: true }), getDoctorCommentsSuccessAction)).toMatchSnapshot();
  expect(doctor(initialDoctorValue.merge({ isLoadingData: true }), getDoctorCommentsErrorAction)).toMatchSnapshot();
})