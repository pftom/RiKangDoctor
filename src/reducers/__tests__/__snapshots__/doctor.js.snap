// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the answer reducer work as well 1`] = `
Immutable.Map {
  doctorInfo: null,
  answers: null,
  comments: null,
  isLoadingData: true,
  loadingError: false,
  loadingSuccess: false,
}
`;

exports[`the answer reducer work as well 2`] = `
Immutable.Map {
  doctorInfo: Immutable.Map {
  },
  answers: null,
  comments: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: true,
}
`;

exports[`the answer reducer work as well 3`] = `
Immutable.Map {
  doctorInfo: null,
  answers: null,
  comments: null,
  isLoadingData: false,
  loadingError: true,
  loadingSuccess: false,
}
`;

exports[`the answer reducer work as well 4`] = `
Immutable.Map {
  doctorInfo: null,
  answers: null,
  comments: null,
  isLoadingData: true,
  loadingError: false,
  loadingSuccess: false,
}
`;

exports[`the answer reducer work as well 5`] = `
Immutable.Map {
  doctorInfo: null,
  answers: Immutable.Map {
  },
  comments: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: true,
}
`;

exports[`the answer reducer work as well 6`] = `
Immutable.Map {
  doctorInfo: null,
  answers: null,
  comments: null,
  isLoadingData: false,
  loadingError: true,
  loadingSuccess: false,
}
`;

exports[`the answer reducer work as well 7`] = `
Immutable.Map {
  doctorInfo: null,
  answers: null,
  comments: null,
  isLoadingData: true,
  loadingError: false,
  loadingSuccess: false,
}
`;

exports[`the answer reducer work as well 8`] = `
Immutable.Map {
  doctorInfo: null,
  answers: null,
  comments: Immutable.Map {
  },
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: true,
}
`;

exports[`the answer reducer work as well 9`] = `
Immutable.Map {
  doctorInfo: null,
  answers: null,
  comments: null,
  isLoadingData: false,
  loadingError: true,
  loadingSuccess: false,
}
`;
